sygnal/sygnal.py

```diff
@@ -52,18 +54,23 @@
 }
 
 
-class Sygnal:
-    def __init__(self, config, custom_reactor, tracer=opentracing.tracer):
+class Sygnal(object):
+    def __init__(
+        self,
+        config: Dict[str, Any],
+        custom_reactor,
+        tracer=opentracing.tracer,
```

Comment:
Are there any obvious annotations for these two we could use?

(Not sure if your goal is "pass mypy" or "pass mypy with `--disallow-untyped-defs`"?)

Reply:
So I struggled with finding an appropriate type for `custom_reactor`. At first I had `asyncioreactor.AsyncioSelectorReactor` which is what is passed in `main`, but this caused problems in `test_utils.py`, where we pass a `ExtendedMemoryReactorClock` instead. I looked in the Twisted docs but couldn't find a reactor that encompassed both these classes so I defaulted to putting nothing in this case. I am sure there is a better solution.

Reply:
You might be able to use one of Twisted's reactor interfaces, like [IReactorCore](https://twistedmatrix.com/documents/current/api/twisted.internet.interfaces.IReactorCore.html). But I suspect that won't work because we probably use methods from more than one interface.

In Sydent I ended up doing this based on what we do in Synapse: https://github.com/matrix-org/sydent/blob/cf27fce07798c9efeea901ddee4211574840024d/sydent/sydent.py#L91-L106

We could try something similar here. Wanna give it a go?